app/src/main/java/org/schabi/newpipe/util/external_communication/ShareUtils.java

```diff
@@ -41,122 +41,111 @@ private ShareUtils() {
      */
     public static void installApp(@NonNull final Context context, final String packageId) {
         // Try market scheme
-        final boolean marketSchemeResult = openIntentInApp(context, new Intent(Intent.ACTION_VIEW,
+        final Intent marketSchemeIntent = new Intent(Intent.ACTION_VIEW,
                 Uri.parse("market://details?id=" + packageId))
-                .setFlags(Intent.FLAG_ACTIVITY_NEW_TASK), false);
-        if (!marketSchemeResult) {
+                .setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        if (!tryOpenIntentInApp(context, marketSchemeIntent)) {
             // Fall back to Google Play Store Web URL (F-Droid can handle it)
-            openUrlInBrowser(context,
-                    "https://play.google.com/store/apps/details?id=" + packageId, false);
+            openUrlInApp(context, "https://play.google.com/store/apps/details?id=" + packageId);
         }
     }
 
     /**
-     * Open the url with the system default browser.
-     * <p>
-     * If no browser is set as default, fallbacks to
-     * {@link #openAppChooser(Context, Intent, boolean)}
+     * Open the url with the system default browser. If no browser is set as default, falls back to
+     * {@link #openAppChooser(Context, Intent, boolean)}. This function selects the package to open
+     * based on which apps respond to the {@code http://} schema alone, which should exclude special
+     * non-browser apps that are can handle the url (e.g. the official YouTube app). Therefore
+     * please <b>prefer {@link #openUrlInApp(Context, String)}</b>, that handles package resolution
+     * in a standard way, unless this is the action of an explicit "Open in browser" button.
      *
-     * @param context                the context to use
-     * @param url                    the url to browse
-     * @param httpDefaultBrowserTest the boolean to set if the test for the default browser will be
-     *                               for HTTP protocol or for the created intent
-     * @return true if the URL can be opened or false if it cannot
-     */
-    public static boolean openUrlInBrowser(@NonNull final Context context,
-                                           final String url,
-                                           final boolean httpDefaultBrowserTest) {
-        final String defaultPackageName;
+     * @param context the context to use
+     * @param url     the url to browse
+     **/
+    public static void openUrlInBrowser(@NonNull final Context context, final String url) {
+        // Resolve using a generic http://, so we are sure to get a browser and not e.g. the yt app.
+        // Note that this requires the `http` schema to be added to `<queries>` in the manifest.
+        final ResolveInfo defaultBrowserInfo = context.getPackageManager().resolveActivity(
+                new Intent(Intent.ACTION_VIEW, Uri.parse("http://")),
+                PackageManager.MATCH_DEFAULT_ONLY);
```

Comment:
Could you solve the deprecation on Android 13 and above, by using the relevant replacement on these Android versions?

Reply:
I was not sure about whether it was really important, considering that we still need to keep the deprecated method, but done